### DISCUSSION TITLE
Display users stars as white instead of in game color

### DISCUSSION THREAD

**Discussion by Skilly146** [Tsangares/stoned_ape_tools#22] [Feature Requests]
## What

Change the users color to white. For example, if the user were green triangle then their stars and carriers would be white triangle instead

## How 

the variable `map.sortedStarSprites[each_user_owned_star _id].ownership` stores an image of that players color/shape combo from the `stars.png` image. The extension should just be able to redefine that variable with an all white version of the users shape. This variable is used in a for loop at line 614 in interface_map.js and is defined in line 335 of interface_map.js with the line `starSprites.ownership = {width: 64, height: 64, pivotX: 32, pivotY: 32, rotation: 0, scale: 1 * map.pixelRatio, image: map.starSrc, spriteX: row * 64, spriteY: col * 64 + 64};`.

**Comment by Tsangares** [comment]
What is the motivation behind this? Why white?

**Reply by Skilly146** [reply]
It doesn't have to be white, that was just an example. The main point is that your stars and carriers should be distinct from any other players. White is good because it matches the intel screen and isn't used for any empire colors. I'm going to add the variable for star colors that I found into the original post

**Reply by Tsangares** [reply]
Okay I like it!

**Reply by anicolao** [reply]
I wanted a similar related thing, the ability to recolour all players arbitrarily. I find same colour/different shape very hard to tell apart, and I'd like to be able to just set the colours for everyone in the game. 

Also could be useful to mark allies.

White for yourself would be a start, but what do people think about making it fully customizable?

**Reply by Skilly146** [reply]
Didn't see your reply @anicolao but that would be pretty easy to implement after adding in a feature like this is, and I also think that would be useful. I think the best way to add that would probably just be copying the Proteus UI buttons almost exactly to an open spot on each players empire screen. There are so many times where you end up allied with or fighting a player who is the same color as you and it sucks

**Reply by anicolao** [reply]
For now, I have implemented a `w` hotkey that toggles the current player between white and their original colour in [NPA](https://github.com/anicolao/npa). The code can handle setting any player to any colour but there is no storage yet to remember those settings so I have yet to do that.

Possibly that makes this FR moot for SAT.
